department-of-veterans-affairs/abd-vro
# EP Merge Prototype

## What was the problem?

Initial Prototype of EP Merge application

## Associated tickets or Slack threads:
- #2063 
- #2044

## How does this fix it?[^1]

* More information can be found in this [mural]. (https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1688079178628/999c88da3e2b95f89b65dea05977157515203ff1?sender=u43ef3ca08acf96162dd44241)
* Keep in mind this prototype is built before changes to the BIP API microservice and BIP API mock service have been updated to include the necessary endpoints. 
* See sequence diagram:
<img width="954" alt="Screenshot 2023-10-10 at 5 56 34 PM" src="https://github.com/department-of-veterans-affairs/abd-vro/assets/135860892/130cfb6c-01b1-4a63-a5eb-ffef836e8a3b">

### Changes Included:
- added ee-ep-merge-app to api-gateway
  - `.github/workflows` added test to api-gateway-integration-test
  - `api-gateway/src/main/resources` - added to api-gateway
  - added env variable to `abd-vro/docker-compose`
- main development `domain-ee/ee-ep-merge-app`:
  - `/integration` integration tests for running with github action `.github/workflows/continous-integration.yml`
  - `/test` unit tests to be run by github action `./github/workflows/test-code.yml` test-python task
  - `/src` source code for application

## How to test this PR
- See README (NOT DONE)


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
